core/editor/src/document/document_message_handler.rs

```diff
@@ -652,13 +658,52 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					}
 				}
 			}
-			FlipLayer(path, flip_horizontal, flip_vertical) => {
-				if let Ok(layer) = self.active_document_mut().document.layer_mut(&path) {
-					let scale = DVec2::new(if flip_horizontal { -1. } else { 1. }, if flip_vertical { -1. } else { 1. });
+			FlipSelectedLayers(axis) => {
+				// TODO: Handle folder nested transforms with the transforms API
+				let selected_paths = self.selected_layers_sorted();
+				if selected_paths.is_empty() {
+					return;
+				}
+
+				let selected_layers = selected_paths.iter().filter_map(|path| {
+					let layer = self.active_document().document.layer(path).unwrap();
+					// TODO: Refactor with `reduce` and `merge_bounding_boxes` once the latter is added
+					let (min, max) = {
+						let bounding_box = layer.bounding_box(layer.transform, layer.style)?;
+						match axis {
+							FlipAxis::X => (bounding_box[0].x, bounding_box[1].x),
+							FlipAxis::Y => (bounding_box[0].y, bounding_box[1].y),
+						}
+					};
+					Some((path.clone(), (min, max)))
+				});
+
+				let (min, max) = selected_layers
+					.clone()
+					.map(|(_, extrema)| extrema)
+					.reduce(|(min_a, max_a), (min_b, max_b)| (min_a.min(min_b), max_a.max(max_b)))
+					.unwrap();
+				let middle = (min + max) / 2.;
+
+				for (path, _) in selected_layers {
+					let layer = self.active_document().document.layer(&path).unwrap();
```

Comment:
Same here.

Reply:
I think not, because this scope is a function with no return value. The `?` operator basically evaluates to:
```rust
let mut f = match f {
    Ok(x) => x,
    Err(e) => return Err(e),
};
```
And we can only return `Err` when the containing function has some `Result` return type.